zilliqa/src/consensus.rs

```diff
@@ -142,6 +144,8 @@ pub struct Consensus {
     transaction_pool: TransactionPool,
     // PRNG - non-cryptographically secure, but we don't need that here
     rng: SmallRng,
+    /// Flag indicating that block creation should be postponed due to empty mempool
+    create_next_block_on_timeout: bool,
```

Comment:
I'm probably missing it, but I can't find the logic that will send the block if we receive a `NewTransaction` while waiting?